### PR TITLE
benchmarks: move crypto benchmarks to criterion

### PR DESCRIPTION
**Pull request by Noah-Kennedy**
Criterion will make it far easier for us to assess performance here.

### CODE REVIEW

**Comment by Noah-Kennedy** [comment]
So, this seems to shed a bit more light on the performance of our custom chacha20 implementation. Good thing @jeff-hiner is getting rid of it.
chacha20

File: boringtun/Cargo.toml [hunk 1]
```diff
@@ -23,13 +23,16 @@ rand_core = "0.5.1"
...
 
 [target.'cfg(target_os="macos")'.dependencies]
 nix = "0.23.1"
 
-[[example]]
-name = "benchmarks"
-path = "src/benchmarks_example.rs"
+[[bench]]
+name = "crypto_benches"
+harness = false
```
Comment (jeff-hiner) [inline boringtun/Cargo.toml:38]: We appear to be missing a newline here.

File: boringtun/benches/crypto_benches/blake2s_benching.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+use boringtun::crypto::Blake2s;
+use criterion::{BenchmarkId, Criterion, Throughput};
+
+pub fn bench_blake2s(c: &mut Criterion) {
+    let mut group = c.benchmark_group("blake2s");
+
+    for size in [128, 1024] {
+        group.throughput(Throughput::Bytes(128 as u64));
```
Comment (jeff-hiner) [inline boringtun/benches/crypto_benches/blake2s_benching.rs:8]: Should this be `size` as well?

File: boringtun/src/ffi/mod.rs [hunk 3]
```diff
@@ -322,16 +319,3 @@ pub unsafe extern "C" fn wireguard_stats(tunnel: *mut Tunn) -> stats {
         reserved: [0u8; 56],
     }
 }
-
-/// Performs an internal benchmark, and returns its result as a C-string.
-#[no_mangle]
-pub extern "C" fn benchmark(name: i32, idx: u32) -> *const c_char {
```
Comment (jeff-hiner) [inline boringtun/src/ffi/mod.rs:328]: Excellent. Let's make sure it's also removed from `wireguard_ffi.h`.

File: boringtun/benches/crypto_benches/blake2s_benching.rs (hunk 2 above)
Comment (Noah-Kennedy) [reply boringtun/benches/crypto_benches/blake2s_benching.rs:8]: ugh, yes

File: boringtun/src/ffi/mod.rs (hunk 3 above)
Comment (Noah-Kennedy) [reply boringtun/src/ffi/mod.rs:328]: We need to, after this, add CI checks for wireguard_ffi.h

File: boringtun/src/wireguard_ffi.h [hunk 4]
```diff
@@ -86,6 +86,4 @@ struct wireguard_result wireguard_force_handshake(struct wireguard_tunnel *tunne
                                                   uint8_t *dst,
                                                   uint32_t dst_size);
 
-struct stats wireguard_stats(struct wireguard_tunnel *tunnel);
-
-const uint8_t *benchmark(int32_t name, uint32_t idx);
+struct stats wireguard_stats(struct wireguard_tunnel *tunnel);
```
Comment (jeff-hiner) [inline boringtun/src/wireguard_ffi.h:89]: gah newline :(

**Review by jeff-hiner — APPROVED** [review]

**Comment by jeff-hiner** [comment]
:ship: :ship: :ship:

**Pull request merged**